src/hooks/useLoginForm.ts

```diff
@@ -1,55 +1,45 @@
-import { useState } from 'react'
+import { useMemo, useState } from 'react'
 
 export function useLoginForm() {
   const [email, setEmail] = useState('')
   const [password, setPassword] = useState('')
-  const [emailError, setEmailError] = useState('')
-  const [passwordError, setPasswordError] = useState('')
   const [emailTouched, setEmailTouched] = useState(false)
   const [passwordTouched, setPasswordTouched] = useState(false)
 
-  const isEmailValid = (value: string) =>
-    /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value)
+  const emailError = useMemo(() => {
+    if (!email) return 'ID를 입력해주세요.'
+    const isEmailValid = (value: string) =>
+      /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value)
+    if (!isEmailValid(email)) return 'ID는 이메일 형식으로 입력해주세요.'
+    return ''
+  }, [email])
 
-  const isPasswordValid = (value: string) => value.length >= 8
+  const passwordError = useMemo(() => {
+    if (!password) return 'PW를 입력해주세요.'
+    const isPasswordValid = (value: string) => value.length >= 8
+    if (!isPasswordValid(password)) return 'PW는 최소 8글자 이상이어야 합니다.'
+    return ''
+  }, [password])
 
-  const validateEmail = (value: string) => {
-    setEmailTouched(true)
-    if (!value) {
-      setEmailError('ID를 입력해주세요.')
-    } else if (!isEmailValid(value)) {
-      setEmailError('ID는 이메일 형식으로 입력해주세요.')
-    } else {
-      setEmailError('')
-    }
-  }
-
-  const validatePassword = (value: string) => {
-    setPasswordTouched(true)
-    if (!value) {
-      setPasswordError('PW를 입력해주세요.')
-    } else if (!isPasswordValid(value)) {
-      setPasswordError('PW는 최소 8글자 이상이어야 합니다.')
-    } else {
-      setPasswordError('')
-    }
-  }
-
-  const isFormValid = isEmailValid(email) && isPasswordValid(password)
+  const isFormValid = useMemo(() => {
+    return !emailError && !passwordError
+  }, [email, password])
 
   return {
-    email,
-    setEmail,
-    password,
-    setPassword,
-    emailError,
-    passwordError,
-    emailTouched,
-    passwordTouched,
-    validateEmail,
-    validatePassword,
-    isFormValid,
-    setEmailTouched,
-    setPasswordTouched,
+    email: {
+      value: email,
+      change: setEmail,
+      error: emailError,
+      touched: emailTouched,
+      onBlur: () => setEmailTouched(true),
+    },
+    password: {
+      value: password,
+      change: setPassword,
+      error: passwordError,
+      touched: passwordTouched,
+      onBlur: () => setPasswordTouched(true),
+    },
+    validForm: isFormValid,
```

Comment:
```suggestion
    const email = useEmail();
    const password = usePassword();
    
    return {
      email,
      password,
      formValid: !email.error && !password.error
    }
```

이렇게 훅을 쪼개서 사용하는 방법도 상상해볼 수 있겠네요 ㅎㅎ

Reply:
수정사항 반영해 코드 수정했습니다.